include/json/value.h

```diff
@@ -895,9 +902,13 @@ class JSON_API ValueIterator : public ValueIteratorBase {
     return *this;
   }
 
-  reference operator*() const { return deref(); }
-
-  pointer operator->() const { return &deref(); }
+  /*! The return value of non-const iterators can be
+   *  changed, so the these functions are not const
+   *  because the returned references/pointers can be used
+   *  to change state of the base class.
+   */
+  reference operator*() { return deref(); }
```

Comment:
This changes an existing symbol, which would require a major version bump if the symbol is publicly accessible.

Would it be possible to keep these 2 operators as `const`? I think it would be ok to *add* 2 non-const operators, if that compiles.

Reply:
Compiles when const

Reply:
I lied.

```
/home/mangix/devstuff/openwrt/build_dir/target-arc_arc700_uClibc/jsoncpp-1.8.4/include/json/value.h:869:45: error: binding reference of type 'Json::ValueIterator::reference' {aka 'Json::Value&'} to 'const Json::Value' discards qualifiers
   reference operator*() const { return deref(); }
```